### DISCUSSION TITLE
run build  error：Fatal JavaScript invalid size error 169220804

### DISCUSSION THREAD

**Discussion by zjj450** [webpack/webpack#18902] [Q&A]
# Bug report

**What is the current behavior?**

when i use this webpack  cache config, npm run build then cause error : Fatal JavaScript invalid size error 169220804：

cache: {
    type: 'filesystem',
    compression: 'gzip',
  }

but i use this config, behavior is normal：

cache: {
    type: 'memory',
    cacheUnaffected: true,
  }

it's dependencies: 
"dependencies": {
    "@babel/core": "^7.20.5",
    "@babel/generator": "^7.20.5",
    "@babel/helper-plugin-utils": "^7.20.2",
    "@babel/parser": "^7.20.5",
    "@babel/plugin-proposal-class-properties": "^7.18.6",
    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.18.6",
    "@babel/plugin-proposal-object-rest-spread": "^7.20.7",
    "@babel/plugin-proposal-optional-chaining": "^7.21.0",
    "@babel/plugin-transform-runtime": "^7.19.6",
    "@babel/preset-env": "^7.19.4",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@babel/traverse": "^7.20.5",
    "@babel/types": "^7.20.5",
    "@builder/pack": "^0.6.5",
    "@builder/webpack-config": "^2.1.1",
    "@iceworks/generate-material": "^1.1.0",
    "@svgr/webpack": "^6.5.1",
    "@swc/helpers": "0.5.0",
    "@types/react": "^18.0.24",
    "@types/react-dom": "^18.0.8",
    "axios": "^1.3.2",
    "babel-core": "^6.26.3",
    "babel-loader": "^9.1.0",
    "case-sensitive-paths-webpack-plugin": "^2.4.0",
    "chalk": "^4.0.0",
    "colors": "^1.4.0",
    "commander": "^9.4.1",
    "convert-source-map": "^2.0.0",
    "create-cli-utils": "^1.0.3",
    "css-loader": "^6.7.1",
    "detect-port": "^1.5.1",
    "download": "^8.0.0",
    "ejs": "^3.1.8",
    "execa": "^5.0.0",
    "f2elint": "^2.2.1",
    "fast-glob": "^3.2.12",
    "file-loader": "^6.2.0",
    "fs-extra": "^11.1.0",
    "get-ip-local": "^1.0.0",
    "html-webpack-plugin": "^5.5.0",
    "inquirer": "^8.2.1",
    "json-stable-stringify": "^1.0.2",
    "less": "^4.1.3",
    "less-loader": "^11.1.0",
    "lodash": "^4.17.21",
    "lodash-es": "^4.17.21",
    "make-error-cause": "^2.3.0",
    "mini-css-extract-plugin": "^2.6.1",
    "openbrowser": "^1.0.2",
    "portfinder": "^1.0.32",
    "postcss": "^8.4.19",
    "postcss-import": "^15.0.0",
    "postcss-loader": "^7.0.1",
    "postcss-plugin-rpx2vw": "^1.0.0",
    "qjsc": "^0.3.0",
    "qrcode-terminal": "^0.12.0",
    "react": "^18.2.0",
    "react-dev-utils": "^12.0.1",
    "sass": "^1.56.1",
    "sass-loader": "^13.2.0",
    "semver": "^7.5.3",
    "source-map": "^0.7.4",
    "style-loader": "^3.3.1",
    "style-unit": "^3.0.5",
    "superagent": "^8.0.9",
    "terser-webpack-plugin": "^5.3.6",
    "ts-loader": "^9.4.2",
    "tsconfig": "^7.0.0",
    "tslib": "^2.4.1",
    "typescript": "^4.9.3",
    "uglify-js": "^3.17.4",
    "url-loader": "^4.1.1",
    "urllib": "^3.17.1",
    "uuid": "^9.0.1",
    "validate-npm-package-name": "^5.0.0",
    "walk": "^2.3.15",
    "webpack": "^5.87.0",
    "webpack-bundle-analyzer": "^4.7.0",
    "webpack-chain": "^6.5.1",
    "webpack-dev-server": "^4.7.4",
    "webpackbar": "^5.0.2",
    "yargs-parser": "^21.1.1"
  },

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

**Other relevant information:**
webpack version:  5.87.0
Node.js version: v18.19.1
Operating System: macOS
Additional tools:

**Comment by Ofc-Pranjal** [comment]
I can help with it

**Comment by alexander-akait** [comment]
Please provide more information - a playable repository or at least a stack, otherwise it will be difficult for us to help

**Comment by zjj450** [comment]
Excuse me , no errors were reported after I deployed the project to the online environment, thank you all
